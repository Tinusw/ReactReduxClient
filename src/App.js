import React, { Component } from "react";
import { Route } from "react-router-dom";

import Header from "./components/header";
import SignIn from "./components/auth/signin";
import SignOut from './components/auth/signout';
import SignUp from './components/auth/signup';
import CampaignIndex from './components/campaign/campaign_index'
import RequireAuth from './components/auth/require_auth'
import Root from './components/root'

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <Route exact path={"/"} component={Root}/>
        <Route path={"/signin"} component={SignIn}/>
        <Route path={"/signout"} component={SignOut}/>
        <Route path={"/signup"} component={SignUp}/>
        <Route path={"/campaign/index"} component={RequireAuth(CampaignIndex)}/>
      </div>
    );
  }
}

export default App;
