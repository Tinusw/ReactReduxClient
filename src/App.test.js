import React from "react";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";

import expect from "expect";
import App from "./App";

// Helpers
import reactElementToJSXString from "react-element-to-jsx-string";

describe("AppComponent", () => {
  const renderer = new ShallowRenderer();
  const actual = renderer.render(<App />);
  const actual_children = actual.props.children;

  it("renders correctly", () => {
    expect(actual).toMatchSnapshot();
  });

  it("has a class of App", () => {
    const expected = "App";
    expect(actual.props.className).toContain(expected);
  });

  it("renders the redux-wrapped Header component", () => {
    expect(
      actual_children.some(
        x =>
          x.type.displayName.includes("Connect(Header)") &&
          x.type.WrappedComponent.toString().includes("Header")
      )
    ).toEqual(true);
  });

  it("renders the root component", () => {
    expect(
      actual_children.some(
        x =>
          x.props.path === "/" && x.props.component.toString().includes("Root")
      )
    ).toEqual(true);
  });

  it("renders the redux-wrapped Signin component", () => {
    expect(
      actual_children.some(
        x =>
          x.props.path === "/signin" &&
          x.props.component.WrappedComponent.toString().includes("SignIn")
      )
    ).toEqual(true);
  });

  it("renders the redux-wrapped SignUp component", () => {
    expect(
      actual_children.some(
        x =>
          x.props.path === "/signup" &&
          x.props.component.WrappedComponent.toString().includes("SignUp")
      )
    ).toEqual(true);
  });

  it("renders the redux-wrapped HOC AUTH component (campaign/index)", () => {
    expect(
      actual_children.some(
        x =>
          x.props.path === "/campaign/index" &&
          x.props.component.WrappedComponent.toString().includes(
            "Authentication"
          )
      )
    ).toEqual(true);
  });

  it("renders all child components", () => {
    expect(actual_children.length).toEqual(6);
  });
});
