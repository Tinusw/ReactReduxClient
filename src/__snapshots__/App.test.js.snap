// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent renders correctly 1`] = `
<div
  className="App"
>
  <Connect(Header) />
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    path="/signin"
  />
  <Route
    component={[Function]}
    path="/signout"
  />
  <Route
    component={[Function]}
    path="/signup"
  />
  <Route
    component={[Function]}
    path="/campaign/index"
  />
</div>
`;
